src/components/common/dialog.tsx

```diff
@@ -0,0 +1,1382 @@
+import { TextInput, type TextInputProps } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert/index.js";
+import {
+    HelperText, HelperTextItem, type HelperTextItemProps
+} from "@patternfly/react-core/dist/esm/components/HelperText";
+import { Checkbox } from "@patternfly/react-core/dist/esm/components/Checkbox";
+import {
+    FormSelect, FormSelectOption, type FormSelectProps,
+} from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
+
+const _ = cockpit.gettext;
+
+type ArrayElement<ArrayType> =
+  ArrayType extends readonly (infer ElementType)[] ? ElementType : never;
+
+interface DialogValidationState {
+    cached_value: unknown;
+    cached_result: string | undefined;
+    timeout_id: number;
+    promise: Promise<void> | undefined;
+    round_id: unknown;
+}
+
+function toSpliced<T>(arr: T[], start: number, deleteCount: number, ...rest: T[]): T[] {
+    const copy = [...arr];
+    copy.splice(start, deleteCount, ...rest);
+    return copy;
+}
+
+export class DialogValue<T> {
+    /* eslint-disable no-use-before-define */
+    #dialog: DialogState<unknown>;
+    /* eslint-enable */
+    #getter: () => T;
+    #setter: (val: T, is_aux: boolean) => void;
+    #path: string;
+
+    constructor(
+        dialog: DialogState<unknown>,
+        getter: () => T,
+        setter: (val: T, is_aux: boolean) => void,
+        path: string
+    ) {
+        this.#dialog = dialog;
+        this.#getter = getter;
+        this.#setter = setter;
+        this.#path = path;
+    }
+
+    validation_text(): string | undefined {
+        return this.#dialog._get_validation(this.#path);
+    }
+
+    get(): T {
+        return this.#getter();
+    }
+
+    set(val: T): void {
+        this.#setter(val, false);
+    }
+
+    set_aux(val: T): void {
+        this.#setter(val, true);
+    }
+
+    id(tag: string = "field"): string {
+        return "dialog-" + tag + "-" + this.#path;
+    }
+
+    map<X>(func: (val: DialogValue<ArrayElement<T>>, index: number) => X): X[] {
+        const val = this.get();
+        if (Array.isArray(val)) {
+            return val.map((_, i) => func(this.sub(i as keyof T) as DialogValue<ArrayElement<T>>, i));
+        } else
+            return [];
+    }
+
+    forEach(func: (val: DialogValue<ArrayElement<T>>, index: number) => void): void {
+        const val = this.get();
+        if (Array.isArray(val)) {
+            val.forEach((_, i) => func(this.sub(i as keyof T) as DialogValue<ArrayElement<T>>, i));
+        }
+    }
+
+    remove(index: number) {
+        const val = this.get();
+        if (Array.isArray(val)) {
+            for (let j = index; j < val.length - 1; j++)
+                this.#dialog._rename_validation_state(this.#path, j + 1, j);
+            this.set(toSpliced(val, index, 1) as T);
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2449-e2dfc49b-20251217-161440-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)